library/Fission/Web/Domain.hs

```diff
@@ -0,0 +1,31 @@
+module Fission.Web.Domain
+  ( API
+  , server
+  ) where
+
+import RIO
+import Data.Has
+
+import           Servant
+
+import           Fission.Web.Server
+import           Fission.User        as User
+import           Fission.IPFS.CID.Types
+
+import qualified Network.AWS.Auth    as AWS
+import qualified Fission.AWS.Types   as AWS
+import qualified Fission.AWS.Route53 as Route53
+
+type API = Capture "cid" CID
+        :> Post    '[PlainText, OctetStream] AWS.Domain
+
+server :: Has AWS.AccessKey cfg
+       => Has AWS.SecretKey cfg
+       => Has AWS.ZoneId    cfg
+       => Has AWS.Domain    cfg
+       => User
+       -> RIOServer         cfg API
+server User { _userID } cid = do
+  let subdomain = User.hashID _userID
+  res <- Route53.registerDomain subdomain cid
+  return res
```

Comment:
I know that this is a WIP, but I'm going to mention it anyways.

The linter will also give you this as a suggestion: any time you see this pattern

```haskell
do
  -- ...
  foo <- runBar
  return foo
```

you can skip the ceremony and just write

```haskell
do
 -- ...
 runBar
```

Reply:
I'm not sure if desugaring makes it clearer or not here, but:

```haskell
runBar >>= \foo -> return foo

-- same as --

runBar
```

GHC is smart enough to rewrite this for you, so there's not runtime impact. One of the advantages of all this pure function stuff is the compiler can be VERY aggressive with rewriting functions to be more efficient ⚡

Reply:
TL;DR

```haskell
server User { _userID } cid = do
  let subdomain = User.hashID _userID
  Route53.registerDomain subdomain cid
```

Could also be:

```haskell
server User { _userID } cid = Route53.registerDomain subdomain cid
  where subdomain = User.hashID _userID
```

But same/same

Reply:
That last version :heart_eyes: 

Also haven't been using a linter :grimacing: Gonna get that set up now. [hlint](https://hackage.haskell.org/package/hlint) seems to be the go to?